src/hatch_openzim/metadata.py

```diff
@@ -0,0 +1,49 @@
+from hatch_openzim.utils import get_github_project_homepage, get_python_versions
+
+
+def update(root: str, config: dict, metadata: dict):
+    """Update the project table's metadata."""
+
+    # Check for absence of metadata we will set + presence in the dynamic property
+    for metadata_key in ["urls", "authors", "keywords", "license", "classifiers"]:
+        if config.get(f"preserve-{metadata_key}", False):
+            # Do not check if we intend to preserve the value set manually
+            continue
+
+        if metadata_key in metadata:
+            raise ValueError(
+                f"'{metadata_key}' may not be listed in the 'project' table when using "
+                "openzim metadata hook."
+            )
+        if metadata_key not in metadata.get("dynamic", []):
+            raise ValueError(
+                f"'{metadata_key}' must be listed in 'project.dynamic' when using "
+                "openzim metadata hook."
+            )
+
+    if not config.get("preserve-urls", False):
+        metadata["urls"] = {
+            "Donate": "https://www.kiwix.org/en/support-us/",
+            "Homepage": get_github_project_homepage(root=root),
+        }
+
+    if not config.get("preserve-authors", False):
+        metadata["authors"] = [{"name": "Kiwix", "email": "dev@kiwix.org"}]
```

Comment:
Might be an opportunity to update those. @kelson42 WDYT?

Reply:
I've opened an issue: https://github.com/openzim/hatch-openzim/issues/2